api/infrastructure/parallelcluster-api.yaml

```diff
@@ -259,23 +262,61 @@ Resources:
                Resource: !GetAtt ParallelClusterFunction.Arn
 
   ParallelClusterFunction:
-    Type: AWS::Serverless::Function
+    Type: AWS::Lambda::Function
     Properties:
-      Tracing: Active
-      PackageType: Image
+      TracingConfig:
+        Mode: Active
       MemorySize: 2048
-      Role: !If [UseCustomParallelClusterFunctionRole, !Ref ParallelClusterFunctionRole, !GetAtt ParallelClusterUserRole.Arn]
+      Timeout: 30
+      Role: !If [UseCustomParallelClusterFunctionRole, !Ref ParallelClusterFunctionRole, !GetAtt [ PclusterPolicies, Outputs.ParallelClusterLambdaRoleArn ]]
       Tags:
-        'parallelcluster:resource': api
-      ImageUri: !If
-        - UseCustomEcrImageUri
-        - !Ref EcrImageUri
-        - !Sub
-          - ${AWS::AccountId}.dkr.ecr.${AWS::Region}.${AWS::URLSuffix}/${Repository}:${Version}
-          - Repository: !Ref PrivateEcrRepository
-            Version: !Join
-              - '-'
-              - [!Select [2, !Split ['/', !Ref EcrImage]], !Select [3, !Split ['/', !Ref EcrImage]]]
+        - Key: 'parallelcluster:resource'
+          Value: api
+        - Key: 'parallelcluster:version'
+          Value: !FindInMap [ParallelCluster, Constants, Version]
+      Runtime: python3.9
+      Handler: index.handler
+      Layers:
+        - !Ref PclusterLayer
+      Code:
```

Comment:
what is the reason why we can't simply use the existing entrypoint rather then inlining it here? is it a limitation of using layers?